### PR TITLE
Precond

### PR DESCRIPTION
**Pull request by GauthamN**
Please describe the purpose of this pull request below.
The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)

Copy/paste from CHANGELOG.md is good

### Added

- this PR merges changes from `0.5.0` release that were merged directly into master (see changelog for details)

### Changed
- Non-persistent state for dev chain (and relevant docs)
- Postcond to assert due to failures
- use new dapp verbose flag

### Removed

- ds-token dependency
- unnecessary logs in `before-tests` script

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,9 +8,7 @@
   "scripts": {
     "lint": "eslint .",
     "oyente": "oyente -s src/Fund.sol",
-    "devchain:before-tests": "parity --chain tests/parity-genesis.config --jsonrpc-apis all",
-    "before-tests": "babel-node tests/before-tests.js",
```
Comment (retotrinkler) [inline package.json]: think above two lines are still needed.
for [setup](https://github.com/melonproject/protocol/tree/develop#setup) instructions in readme

File: src/assets/AssetRegistrar.sol [hunk 2]
```diff
@@ -25,6 +25,9 @@ contract AssetRegistrar is DBC, Owned, AssetRegistrarInterface {
 
     // FIELDS
 
+    // Constant fields
+    uint constant MAX_ASSETS = 50; // Max number of registrable assets
```
Comment (retotrinkler) [inline src/assets/AssetRegistrar.sol]: nice

**Comment by travs** [comment]
Fixing this build right now

File: package.json (hunk 1 above)
Comment (GauthamN) [reply package.json]: Updating the readme

**Comment by travs** [comment]
For reference when we merge precond into develop, it will also include all the changes from 0.5.0 that went straight to master

**Pull request merged**